### ISSUE TITLE
[Bug]: Dropdown in JSON Form doesn't close on clicking outside the dropdown

### ISSUE THREAD

**Issue by aswathkk** [appsmithorg/appsmith#12910]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

https://user-images.githubusercontent.com/10436935/163262358-9c9f0b02-dbe5-42fa-91ec-69f88adcb869.mov

Clicking outside the dropdown doesn't close the dropdown consistently. This is observed only in edit mode.

### Steps To Reproduce

1. Add a JSON Form widget
2. Make one field to be a dropdown
3. open the dropdown
4. Try clicking outside the dropdown

### Version

Cloud v1.6.20

**Comment by somangshu** [comment]
I observed a very similar behaviour on a dashboard I too was working on, This seems like an issue with the select / multiselect issue, maybe its not really related to the JSON form. 

https://user-images.githubusercontent.com/11089579/163334064-8eca814a-ecce-47bb-ba64-21c8ff79876a.mov

[Chat ref](https://theappsmith.slack.com/archives/C02JV8G1MP0/p1648720297452199)

**Comment by Tooluloope** [comment]
@somangshu this isn't a Select or MultiSelect issue

@ashit-rath, I noticed the dropdown doesn't readjust after the first render as seen in the Recording.

https://user-images.githubusercontent.com/31691737/163948639-9408246c-49bc-4825-8696-6d93791e12be.mov

**Comment by ashit-rath** [comment]
@Tooluloope Is right, it's not a select or multiselect issue,  it's actually the byproduct of a feature.
We have label-input association, eg - If you have text input field and you click on the label rather than the input, the input would be focused.

As our widgets didn't support ids we had to use the explicit label-input association.

https://css-tricks.com/html-inputs-and-labels-a-love-story/

The exact same behaviour should be present in our widgets as well. I think we should take this up in the widget reskinning and use the recommended way to have labels and inputs.

**Comment by ashit-rath** [comment]
@Tooluloope Yes you are right about the dropdown width; it was done for some performance reasons and I was thinking of dealing with it separately when dealing with the select /  multi select perf.
The App viewers won't see this issue thought.

**Comment by Tooluloope** [comment]
@ashit-rath That's fine, But I guess we might need to handle the flickering of the dropdown. So it doesn't flicker when the Label is clicked

**Comment by dilippitchika** [comment]
@ashit-rath i guess the problem is flickering of dropdown
